Tests/ML/datasets/test_dataset.py

```diff
@@ -419,6 +421,74 @@ def test_get_all_metadata(default_config: ModelConfigBase) -> None:
+        ["3",  "train_and_test_data/id2_channel1.nii.gz", "channel2",  "3"],
+        ["3",  "train_and_test_data/id2_mask.nii.gz",     "mask",      "3"],
+        ["3",  "train_and_test_data/id2_region.nii.gz",   "region",    "3"],
+        ["3",  "train_and_test_data/id2_region.nii.gz",   "region_1",  "3"],
+        ["4",  "train_and_test_data/id2_channel1.nii.gz", "channel1",  "3"],
+        ["4",  "train_and_test_data/id2_channel1.nii.gz", "channel2",  "3"],
+        ["4",  "train_and_test_data/id2_mask.nii.gz",     "mask",      "3"],
+        ["4",  "train_and_test_data/id2_region.nii.gz",   "region",    "3"],
+        ["4",  "train_and_test_data/id2_region.nii.gz",   "region_1",  "3"],
+        ["5",  "train_and_test_data/id2_channel1.nii.gz", "channel1",  "3"],
+        ["5",  "train_and_test_data/id2_channel1.nii.gz", "channel2",  "3"],
+        ["5",  "train_and_test_data/id2_mask.nii.gz",     "mask",      "3"],
+        ["5",  "train_and_test_data/id2_region.nii.gz",   "region",    "3"],
+        ["5",  "train_and_test_data/id2_region.nii.gz",   "region_1",  "3"],
+        ["6",  "train_and_test_data/id2_channel1.nii.gz", "channel1",  "3"],
+        ["6",  "train_and_test_data/id2_channel1.nii.gz", "channel2",  "3"],
+        ["6",  "train_and_test_data/id2_mask.nii.gz",     "mask",      "3"],
+        ["6",  "train_and_test_data/id2_region.nii.gz",   "region",    "3"],
+        ["6",  "train_and_test_data/id2_region.nii.gz",   "region_1",  "3"],
+        # ["7",  "train_and_test_data/id2_channel1.nii.gz", "channel1",  "4"],
+        # ["7",  "train_and_test_data/id2_channel1.nii.gz", "channel2",  "4"],
+        # ["7",  "train_and_test_data/id2_mask.nii.gz",     "mask",      "4"],
+        ["7",  "train_and_test_data/id2_region.nii.gz",   "region",    "4"],
+        ["7",  "train_and_test_data/id2_region.nii.gz",   "region_1",  "4"]]
+
+    config_missing_channels_and_files = DummyModel()
+    data_frame_with_missing_channels_and_files = pd.DataFrame(data_frame,
+                                                             columns=['subject','filePath','channel','institutionId'])
+    config_missing_channels_and_files._dataset_data_frame = data_frame_with_missing_channels_and_files
+    container_missing_files_channels = InnerEyeContainer(config_missing_channels_and_files)
+    with pytest.raises(ValueError) as e:
+        container_missing_files_channels.setup()
+        
+    assert "Patient 1 does not have channel 'channel1'" in str(e.value)
+    assert "Patient 1 does not have channel 'channel2'" in str(e.value)
+    assert "Patient 1 does not have channel 'mask'" in str(e.value)
+    assert "Patient 2" in str(e.value) and "FILE_A does not exist" in str(e.value)
+    assert "Patient 7 does not have channel 'channel1'" in str(e.value)
+    assert "Patient 7 does not have channel 'channel2'" in str(e.value)
+    assert "Patient 7 does not have channel 'mask'" in str(e.value)
```

Comment:
asserts only cover patients 1, 2, 7. Data file also have patients 3, 4, 6. It should be enough to have a test case with 1 fully valid patient, and several invalid ones.
Also, the asserts should say that "Patient 6" is NOT in the error message.

Reply:
Simplified unit test to include patients 1,2,3,4. Asserted case for working patient (3)  and not working patients (1,2,4).